### PR TITLE
Preparation for additional contributors

### PR DESCRIPTION
**Pull request by thirtytwobits**
(I already release the build container to dockerhub)

### CODE REVIEW

**Comment by SyrianSpock** [comment]
Not using tox to run tests?

**Comment by thirtytwobits** [comment]
> Not using tox to run tests?

I'm unsure what you are asking. 

> Are we not using tox? 

Yes we are, in our Python projects like Nunavut.

> Are we using tox to run c++/googletest tests? 

Not really. Tox is for Python. We're using it in Nanaimo's c++ verification build to create a consistent Python environment for nnvg. So we use tox to ensure that python is available to the test process and that nnvg and its dependencies are installed. We then use nnvg to generate C++ types and use googletest to run tests against that generated, native code.

**Comment by SyrianSpock** [comment]
> I'm unsure what you are asking.

I was asking because the PR only modifies the Docker image.

> Yes we are, in our Python projects like Nunavut.

Ok, I didn't know.

> Not really. Tox is for Python. We're using it in Nanaimo's c++ verification build to create a consistent Python environment for nnvg. So we use tox to ensure that python is available to the test process and that nnvg and its dependencies are installed. We then use nnvg to generate C++ types and use googletest to run tests against that generated, native code.

Souns a bit elaborate, but OK, fair enough, if it works :shipit:

**Comment by thirtytwobits** [comment]
> Souns a bit elaborate, but OK, fair enough, if it works :shipit:

Yeah, that's a fair concern. I'm more concerned with using global python environments which might try to pull the source under test from Pypi rather then testing what's in the container. Using virtualenv solves this. Using tox automates setting up the virtualenv (i.e. I was able to remove CMake code by adding tox).

**Comment by thirtytwobits** [comment]
Sorry, this got mixed with a pending PR. I need to figure out why my builds are failing. Looks like an infrastructure problem.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+Despite the fact that the library itself can be used on virtually any platform that has a standard-compliant
+C++11 compiler, the library development process assumes that the host OS is Linux or OSX.
+
+Prerequisites:
+
+* Google test library for C++ - gtest (downloaded as part of the build from [github](https://github.com/google/googletest))
+* C++11 capable compiler with GCC-like interface (e.g. GCC, Clang)
+* CMake 3.5+
+* clang-format
```
Comment (pavel-kirienko) [inline CONTRIBUTING.md]:
```suggestion
* clang-format
* clang-tidy
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+1. libuavcan v1 favors idiomatic C++ over defining its own utilities and helpers.
+1. libuavcan v1 is a header-only library.
+
+## About the Blue Sky effort
+
+You should never rewrite a codebase from scratch. It's a terrible idea. We're rewriting libuavcan from scratch. We're terrible developers.
+
+Sorry.
+
+The reason we opted for a complete rewrite is that so much is changing. Between the updates to the specification, the abandonment of C++98 support, switching to header-only, removing the drivers from the main repository, adding CAN FD support, etc, etc, etc; it was obvious that we'd be rewriting everything anyway. The good news is that v0 exists, is fully supported, and will be liberally copy-and-pasted from as this makes sense for v1. What we don't have is any git history tracing from v1 since this would be deceiving. Futhermore, the unit tests in uavcan v0 are a bit of a mess so we'll be writing them in a way that is more sustainable.
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+**Libuavcan source should be fluent. Comment everything in plain prose, build the docs with each change, read the docs to make sure your comments make sense.**
+
+Despite the fact that the library itself can be used on virtually any platform that has a standard-compliant
+C++11 compiler, the library development process assumes that the host OS is Linux or OSX.
+
+Prerequisites:
+
+* Google test library for C++ - gtest (downloaded as part of the build from [github](https://github.com/google/googletest))
+* C++11 capable compiler with GCC-like interface (e.g. GCC, Clang)
```
[the remark this answers is not in the document]
Comment (thirtytwobits) [reply CONTRIBUTING.md]: sure.

**Pull request merged**